src/bundle/Resources/public/js/scripts/admin.contenttype.matrix.js

```diff
@@ -37,12 +41,28 @@
     };
     const removeItems = (event) => {
         const settingsNode = event.target.closest(SELECTOR_SETTINGS_COLUMNS);
+        const checkedRows = findCheckedColumns(settingsNode);
+        const allRows = settingsNode.querySelectorAll(':not(.ibexa-table__template--empty-row) .ibexa-table__row');
+        const tableHead = settingsNode.querySelector('.ibexa-table thead');
+        const tableBody = settingsNode.querySelector('.ibexa-table__body');
+        const emptyPlaceholder = settingsNode.querySelector('.ibexa-table__template--empty-row')?.content.cloneNode(true);
+        const removedAllRows = checkedRows.length === allRows.length;
 
-        findCheckedColumns(settingsNode).forEach((checkbox) => {
-            checkbox.checked = false;
-            checkbox.dispatchEvent(new Event('change'));
-            checkbox.closest(SELECTOR_COLUMN).remove();
-        });
+        if (removedAllRows) {
+            tableBody.appendChild(emptyPlaceholder);
+        }
+
+        setTimeout(() => {
```

Comment:
This timeout is necessary as without it hiding/removing elements happens to fast, before returning emptyPlaceholder and for short moment there's no element and whole table "jumps"